// react-rainbow-components/components/FileSelector/styled/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TruncatedText = exports.StyledInput = exports.StyledIconContainer = exports.StyledDropzone = exports.StyledContainer = exports.StyledButtonIcon = exports.StyledBackdrop = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _attachThemeAttrs = _interopRequireDefault(require("../../../styles/helpers/attachThemeAttrs"));

var _iconContainer = _interopRequireDefault(require("../../Input/styled/iconContainer"));

var _ButtonIcon = _interopRequireDefault(require("../../ButtonIcon"));

var _margins = require("../../../styles/margins");

var _borderRadius = require("../../../styles/borderRadius");

var _fontSizes = require("../../../styles/fontSizes");

var _paddings = require("../../../styles/paddings");

var _replaceAlpha = _interopRequireDefault(require("../../../styles/helpers/color/replaceAlpha"));

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7;

var StyledContainer = _styledComponents["default"].div(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n    display: flex;\n    flex-direction: column;\n    align-items: center;\n    justify-content: center;\n    width: 100%;\n    height: 100%;\n"])));

exports.StyledContainer = StyledContainer;
var StyledDropzone = (0, _attachThemeAttrs["default"])(_styledComponents["default"].div)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["     \n    position: relative;\n    width: 100%;\n    height: 2.5rem;\n    padding: 4px;\n    border: dashed 1px ", ";\n    border-radius: ", ";\n    background-color: ", ";\n    color: ", ";\n    display: flex;\n    flex-direction: column;\n    justify-content: center;\n    align-items: center; \n\n    &:hover {\n        cursor: pointer;\n        border: dashed 1px ", ";\n    }\n\n    ", "\n\n    ", "\n\n    ", "\n\n    ", "\n    \n    ", "\n\n\n    ", ";\n\n    ", ";\n\n    ", ";\n\n"])), function (props) {
  return props.palette.text.disabled;
}, _borderRadius.BORDER_RADIUS_2, function (props) {
  return props.palette.background.disabled;
}, function (props) {
  return props.palette.text.main;
}, function (props) {
  return props.palette.brand.main;
}, function (props) {
  return props.hasFocus && "\n        outline: 0;\n        border: solid 1px ".concat(props.palette.brand.main, ";\n        box-shadow: ").concat(props.shadows.brand, ";\n    ");
}, function (props) {
  return props.variant === 'multiline' && "\n        min-height: 12rem;\n        border-radius: 27px;\n        height: 100%;\n        width: 100%;\n    ";
}, function (props) {
  return props.disabled && "\n        color: ".concat(props.palette.text.disabled, ";\n\n        &:hover {\n            cursor: not-allowed;\n            border-color: ").concat(props.palette.text.disabled, ";\n        }\n    ");
}, function (props) {
  return props.error && "\n        color: ".concat(props.palette.error.main, ";\n        border: dashed 1px ").concat(props.palette.error.main, ";\n        background-color: ").concat((0, _replaceAlpha["default"])(props.palette.error.main, 0.1), ";\n\n        &:hover {\n            border: dashed 1px ").concat(props.palette.error.main, ";\n        }\n\n        ").concat(props.hasFocus && "\n            outline: 0;\n            border: solid 1px ".concat(props.palette.error.main, ";\n            box-shadow: ").concat(props.shadows.error, ";\n            "), "\n        ");
}, function (props) {
  return props.isDragOver && "\n        color: ".concat(props.palette.brand.main, ";\n        background-color: ").concat((0, _replaceAlpha["default"])(props.palette.brand.main, 0.05), ";\n        border-color: ").concat(props.palette.brand.main, ";\n    ");
}, function (props) {
  return props.borderRadius === 'square' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SQUARE, ";\n        ");
}, function (props) {
  return props.borderRadius === 'semi-square' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SEMI_SQUARE, ";\n        ");
}, function (props) {
  return props.borderRadius === 'semi-rounded' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SEMI_ROUNDED, ";\n        ");
});
exports.StyledDropzone = StyledDropzone;

var TruncatedText = _styledComponents["default"].span(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n    white-space: nowrap;\n    overflow: hidden;\n    text-overflow: ellipsis;\n    max-width: 100%;\n    pointer-events: none;\n"])));

exports.TruncatedText = TruncatedText;
var StyledBackdrop = (0, _attachThemeAttrs["default"])(_styledComponents["default"].div)(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2["default"])(["\n    width: 100%;\n    height: 100%;\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    padding-left: 2.35rem;\n    padding-right: ", ";\n    border-radius: ", ";\n    pointer-events: none;\n\n    ", "\n\n    ", "\n    \n    ", ";\n\n    ", ";\n\n    ", ";\n"])), _paddings.PADDING_MEDIUM, _borderRadius.BORDER_RADIUS_2, function (props) {
  return props.isFileSelected && "\n        justify-content: left;\n        background-color: ".concat(props.palette.background.highlight, ";\n        padding-right: 2.35rem;\n        ");
}, function (props) {
  return props.variant === 'multiline' && "\n        flex-direction: column;\n        justify-content: center;\n        align-items: center;\n        text-align: center;\n        font-size: ".concat(_fontSizes.FONT_SIZE_TEXT_MEDIUM, ";\n        padding: ").concat(_paddings.PADDING_MEDIUM, ";\n        background: transparent;\n        height: 100%;\n        width: 100%;\n        ");
}, function (props) {
  return props.borderRadius === 'square' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SQUARE, ";\n        ");
}, function (props) {
  return props.borderRadius === 'semi-square' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SEMI_SQUARE, ";\n        ");
}, function (props) {
  return props.borderRadius === 'semi-rounded' && "\n            border-radius: ".concat(_borderRadius.BORDER_RADIUS_SEMI_ROUNDED, ";\n        ");
});
exports.StyledBackdrop = StyledBackdrop;
var StyledIconContainer = (0, _attachThemeAttrs["default"])((0, _styledComponents["default"])(_iconContainer["default"]))(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2["default"])(["\n    svg {\n        width: 24px !important;\n        height: 24px !important;\n        font-size: 24px !important;\n        color: ", ";\n\n        ", "\n\n        ", "\n    }\n\n    ", "\n\n    ", "\n    ", "\n\n    ", "\n"])), function (props) {
  return props.palette.brand.main;
}, function (props) {
  return props.error && "\n            color: ".concat(props.palette.error.main, "\n        ");
}, function (props) {
  return props.disabled && "\n            color: ".concat(props.palette.text.disabled, ";\n        ");
}, function (props) {
  return props.isSingleFile && props.variant === 'inline' && "\n        svg {\n            width: 18px !important;\n            height: 18px !important;\n            font-size: 18px !important;\n        }\n    ";
}, function (props) {
  return props.iconPosition === 'left' && "\n        left: ".concat(props.readOnly ? 0 : '0.8rem', ";\n    ");
}, function (props) {
  return props.iconPosition === 'right' && "\n        right: ".concat(props.readOnly ? 0 : '0.3rem', ";\n    ");
}, function (props) {
  return props.variant === 'multiline' && "\n        position: static;\n        width: 64px;\n        height: auto;\n        margin-bottom: ".concat(_margins.MARGIN_X_SMALL, ";\n        svg {\n            width: 64px !important;\n            height: 64px !important;\n            font-size: 64px !important;\n\n            ").concat(props.error && "\n                height: 40px !important;\n                width: 40px !important;\n                font-size: 40px !important;\n                color: ".concat(props.palette.error.main, ";\n            "), "\n        }\n    ");
});
exports.StyledIconContainer = StyledIconContainer;
var StyledButtonIcon = (0, _styledComponents["default"])(_ButtonIcon["default"])(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2["default"])(["\n    width: unset;\n    height: unset;\n    pointer-events: auto;\n    margin-right: 6px;\n    ", "\n\n    svg {\n        pointer-events: none;\n    }\n"])), function (props) {
  return props.pickerVariant === 'multiline' && "\n        position: absolute;\n        top: 12px;\n        right: 2px;\n    ";
});
exports.StyledButtonIcon = StyledButtonIcon;

var StyledInput = _styledComponents["default"].input(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2["default"])(["\n    position: absolute;\n    top: 0;\n    left: 0;\n    width: 100%;\n    height: 100%;\n    opacity: 0;\n\n    &:hover {\n        cursor: pointer;\n    }\n\n    ", "\n"])), function (props) {
  return props.disabled && "\n        &:hover {\n            cursor: not-allowed;\n        }\n    ";
});

exports.StyledInput = StyledInput;
