// react-rainbow-components/styles/shadows.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SHADOW_OUTLINE_2 = exports.SHADOW_OUTLINE = exports.SHADOW_ERROR = exports.SHADOW_6 = exports.SHADOW_5 = exports.SHADOW_4 = exports.SHADOW_3 = exports.SHADOW_2 = exports.SHADOW_1 = void 0;

var _colors = require("./colors");

var SHADOW_OUTLINE = "0 0 2px ".concat(_colors.COLOR_BRAND);
exports.SHADOW_OUTLINE = SHADOW_OUTLINE;
var SHADOW_OUTLINE_2 = "0 0 8px ".concat(_colors.COLOR_BRAND);
exports.SHADOW_OUTLINE_2 = SHADOW_OUTLINE_2;
var SHADOW_ERROR = "0 0 2px ".concat(_colors.COLOR_ERROR);
exports.SHADOW_ERROR = SHADOW_ERROR;
var SHADOW_1 = "0 0 2px 0 ".concat(_colors.COLOR_GRAY_3);
exports.SHADOW_1 = SHADOW_1;
var SHADOW_2 = "0 2px 4px 0 ".concat(_colors.COLOR_GRAY_2);
exports.SHADOW_2 = SHADOW_2;
var SHADOW_3 = "0 0 1px 0 ".concat(_colors.COLOR_GRAY_3);
exports.SHADOW_3 = SHADOW_3;
var SHADOW_4 = "0 1px 2px 0 ".concat(_colors.COLOR_GRAY_2);
exports.SHADOW_4 = SHADOW_4;
var SHADOW_5 = "0 0 3px ".concat(_colors.COLOR_GRAY_1);
exports.SHADOW_5 = SHADOW_5;
var SHADOW_6 = "0 2px 12px 0 ".concat(_colors.COLOR_GRAY_2);
exports.SHADOW_6 = SHADOW_6;
