// react-rainbow-components/components/ImportRecordsFlow/stepFour/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = StepFour;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Column = _interopRequireDefault(require("../../Column"));

var _getPreviewDataToImport = _interopRequireDefault(require("../helpers/getPreviewDataToImport"));

var _table = _interopRequireDefault(require("../styled/table"));

function StepFour(props) {
  var schemaFields = props.schemaFields,
      data = props.data,
      fieldsMap = props.fieldsMap,
      attributes = props.attributes;
  var previewData = (0, _getPreviewDataToImport["default"])(data.slice(0, 5), fieldsMap, attributes);
  return _react["default"].createElement(_table["default"], {
    keyField: "id",
    data: previewData
  }, schemaFields.map(function (field, index) {
    var key = "column-".concat(index);
    return _react["default"].createElement(_Column["default"], {
      key: key,
      header: field,
      field: field
    });
  }));
}

StepFour.propTypes = {
  schemaFields: _propTypes["default"].array,
  data: _propTypes["default"].array,
  fieldsMap: _propTypes["default"].object,
  attributes: _propTypes["default"].object
};
StepFour.defaultProps = {
  schemaFields: [],
  data: [],
  fieldsMap: {},
  attributes: {}
};
